### PR TITLE
Make safe init checker skip global objects

### PR DESCRIPTION
**Pull request by q-ata**
Decouple the checking of classes and global objects. The `Ysafe-init-global` flag can be used for checking global objects instead.

### CODE REVIEW

File: tests/init/neg/i12544.scala [hunk 1]
```diff
@@ -2,18 +2,23 @@ enum Enum:
...
+    val b: Enum = f((new nested).a)
 
-  def f(e: Enum): Enum = e
+    def f(e: Enum): Enum = e
 
-  class B() { def foo() = n + 1 }
-  g(new B())                       // error
-  val n: Int = 10
+    class B() { def foo() = n + 1 }
+    g(new e.B())
```
Comment (olhotak) [inline tests/init/neg/i12544.scala:18]: Is there still an error here? Should there be?

File: tests/init/neg/i15883.scala [hunk 2]
```diff
@@ -1,2 +1,4 @@
-val a = b
```
Comment (olhotak) [inline tests/init/neg/i15883.scala:1]: As we discussed, move the original test to the global init tests.

**Review by olhotak — APPROVED** [review]
Otherwise LGTM.

**Review by liufengyun — APPROVED** [review]

File: tests/init/neg/i12544.scala (hunk 1 above)
Comment (olhotak) [reply tests/init/neg/i12544.scala:18]:
```suggestion
    g(new B())
```

**Comment by liufengyun** [comment]
A rebase is need to update the check fiiles due to #18930.

**Pull request merged**